src/main/java/gregtech/api/metatileentity/multiblock/RecipeMapSteamMultiblockController.java

```diff
@@ -99,38 +99,41 @@ private void resetTileAbilities() {
-            }
-
-            if (!recipeMapWorkable.isWorkingEnabled()) {
-                textList.add(new TextComponentTranslation("gregtech.multiblock.work_paused"));
-
-            } else if (recipeMapWorkable.isActive()) {
-                textList.add(new TextComponentTranslation("gregtech.multiblock.running"));
-                int currentProgress = (int) (recipeMapWorkable.getProgressPercent() * 100);
-                if (this.recipeMapWorkable.getParallelLimit() != 1) {
-                    textList.add(new TextComponentTranslation("gregtech.multiblock.parallel", this.recipeMapWorkable.getParallelLimit()));
-                }
-                textList.add(new TextComponentTranslation("gregtech.multiblock.progress", currentProgress));
-            } else {
-                textList.add(new TextComponentTranslation("gregtech.multiblock.idling"));
-            }
-        }
+        MultiblockDisplayText.builder(textList, isStructureFormed())
+                .setWorkingStatus(recipeMapWorkable.isWorkingEnabled(), recipeMapWorkable.isActive())
+                .addCustom(tl -> {
+                    // custom steam tank line
+                    IFluidTank steamFluidTank = recipeMapWorkable.getSteamFluidTankCombined();
+                    if (steamFluidTank != null && steamFluidTank.getCapacity() > 0) {
+                        String stored = TextFormattingUtil.formatNumbers(steamFluidTank.getFluidAmount());
+                        String capacity = TextFormattingUtil.formatNumbers(steamFluidTank.getCapacity());
+
+                        ITextComponent steamInfo = TextComponentUtil.stringWithColor(
+                                TextFormatting.BLUE,
+                                stored + " / " + capacity + " L");
+
+                        tl.add(TextComponentUtil.translationWithColor(
+                                TextFormatting.GRAY,
+                                "gregtech.multiblock.steam.steam_stored",
+                                steamInfo));
+                    }
+                })
+                .addParallelsLine(recipeMapWorkable.getParallelLimit())
+                .addWorkingStatusLine()
+                .addProgressLine(recipeMapWorkable.getProgressPercent());
     }
 
     @Override
     protected void addWarningText(List<ITextComponent> textList) {
-        super.addWarningText(textList);
-        if (isStructureFormed()) {
-            if (recipeMapWorkable.isHasNotEnoughEnergy()) {
-                textList.add(new TextComponentTranslation("gregtech.multiblock.steam.low_steam").setStyle(new Style().setColor(TextFormatting.RED)));
-            }
-        }
+        MultiblockDisplayText.builder(textList, isStructureFormed(), false)
+                .addCustom(tl -> {
+                    if (isStructureFormed() && recipeMapWorkable.isHasNotEnoughEnergy()) {
+                        tl.add(TextComponentUtil.translationWithColor(
```

Comment:
It works like the previous warning, and like the energy warning, so it only displays while a recipe is actively running. This would be a larger change in recipe logic to make that different but I'm not sure if its necessary anyway